src/transformers/feature_extraction_utils.py

```diff
@@ -382,6 +381,22 @@ def save_pretrained(self, save_directory: Union[str, os.PathLike], push_to_hub:
             kwargs (`Dict[str, Any]`, *optional*):
                 Additional key word arguments passed along to the [`~utils.PushToHubMixin.push_to_hub`] method.
         """
+        token = kwargs.pop("token", None)
+        use_auth_token = kwargs.pop("use_auth_token", None)
+
+        if use_auth_token is not None:
+            warnings.warn(
+                "The `use_auth_token` argument is deprecated and will be removed in v5 of Transformers.", FutureWarning
+            )
+            if token is not None:
```

Comment:
```suggestion
            if kwargs.get("token", None) is not None:
```

Reply:
Changed to (for which where `token` is not explicitly in the arguments)

```python
        use_auth_token = kwargs.pop("use_auth_token", None)

        if use_auth_token is not None:
            warnings.warn(
                "The `use_auth_token` argument is deprecated and will be removed in v5 of Transformers.", FutureWarning
            )
            if kwargs.get("token", None) is not None:
                raise ValueError(
                    "`token` and `use_auth_token` are both specified. Please set only the argument `token`."
                )
            kwargs["token"] = use_auth_token
```

Reply:
Latter, when we want to add `token` explicitly in the arguments of other methods (`save_pretrained` for example), we will need modify this block.